<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>librepcb::editor::BoardDesignRulesDialog</class>
 <widget class="QDialog" name="librepcb::editor::BoardDesignRulesDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>613</width>
    <height>252</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Board Design Rules</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="0" column="2">
    <widget class="QLabel" name="label_5">
     <property name="text">
      <string>Ratio (% of diam.)</string>
     </property>
    </widget>
   </item>
   <item row="6" column="2">
    <widget class="librepcb::editor::UnsignedRatioEdit" name="edtPadAnnularRingRatio" native="true"/>
   </item>
   <item row="5" column="0">
    <widget class="QLabel" name="label_2">
     <property name="text">
      <string>Inner Layer Pads:</string>
     </property>
    </widget>
   </item>
   <item row="3" column="1">
    <widget class="librepcb::editor::UnsignedLengthEdit" name="edtSolderPasteClrMin" native="true"/>
   </item>
   <item row="2" column="3">
    <widget class="librepcb::editor::UnsignedLengthEdit" name="edtStopMaskClrMax" native="true"/>
   </item>
   <item row="7" column="3">
    <widget class="librepcb::editor::UnsignedLengthEdit" name="edtViaAnnularRingMax" native="true"/>
   </item>
   <item row="7" column="2">
    <widget class="librepcb::editor::UnsignedRatioEdit" name="edtViaAnnularRingRatio" native="true"/>
   </item>
   <item row="8" column="0" colspan="4">
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Apply|QDialogButtonBox::Cancel|QDialogButtonBox::Ok|QDialogButtonBox::RestoreDefaults</set>
     </property>
    </widget>
   </item>
   <item row="1" column="0">
    <widget class="QLabel" name="label_6">
     <property name="text">
      <string>Stop Mask Max. Via Diam.:</string>
     </property>
    </widget>
   </item>
   <item row="6" column="3">
    <widget class="librepcb::editor::UnsignedLengthEdit" name="edtPadAnnularRingMax" native="true"/>
   </item>
   <item row="1" column="3">
    <widget class="librepcb::editor::UnsignedLengthEdit" name="edtStopMaskMaxViaDia" native="true"/>
   </item>
   <item row="6" column="1">
    <widget class="librepcb::editor::UnsignedLengthEdit" name="edtPadAnnularRingMin" native="true"/>
   </item>
   <item row="3" column="0">
    <widget class="QLabel" name="label_8">
     <property name="text">
      <string>Solder Paste Clearance:</string>
     </property>
    </widget>
   </item>
   <item row="3" column="3">
    <widget class="librepcb::editor::UnsignedLengthEdit" name="edtSolderPasteClrMax" native="true"/>
   </item>
   <item row="7" column="0">
    <widget class="QLabel" name="label_10">
     <property name="text">
      <string>Vias Annular Ring:</string>
     </property>
    </widget>
   </item>
   <item row="0" column="3">
    <widget class="QLabel" name="label_9">
     <property name="text">
      <string>Maximum</string>
     </property>
    </widget>
   </item>
   <item row="3" column="2">
    <widget class="librepcb::editor::UnsignedRatioEdit" name="edtSolderPasteClrRatio" native="true"/>
   </item>
   <item row="2" column="1">
    <widget class="librepcb::editor::UnsignedLengthEdit" name="edtStopMaskClrMin" native="true"/>
   </item>
   <item row="0" column="1">
    <widget class="QLabel" name="label_4">
     <property name="text">
      <string>Minimum</string>
     </property>
    </widget>
   </item>
   <item row="2" column="0">
    <widget class="QLabel" name="label_3">
     <property name="text">
      <string>Stop Mask Clearance:</string>
     </property>
    </widget>
   </item>
   <item row="2" column="2">
    <widget class="librepcb::editor::UnsignedRatioEdit" name="edtStopMaskClrRatio" native="true"/>
   </item>
   <item row="7" column="1">
    <widget class="librepcb::editor::UnsignedLengthEdit" name="edtViaAnnularRingMin" native="true"/>
   </item>
   <item row="6" column="0">
    <widget class="QLabel" name="label_7">
     <property name="text">
      <string>Autom. Pads Annular Ring:</string>
     </property>
    </widget>
   </item>
   <item row="4" column="0">
    <widget class="QLabel" name="label">
     <property name="text">
      <string>Component Side Pads:</string>
     </property>
    </widget>
   </item>
   <item row="4" column="1" colspan="3">
    <layout class="QHBoxLayout" name="horizontalLayout_2">
     <item>
      <widget class="QRadioButton" name="rbtnCmpSidePadFullShape">
       <property name="text">
        <string>Full Shape</string>
       </property>
       <property name="checked">
        <bool>true</bool>
       </property>
       <attribute name="buttonGroup">
        <string notr="true">rbtnGroupCmpSidePadShape</string>
       </attribute>
      </widget>
     </item>
     <item>
      <widget class="QRadioButton" name="rbtnCmpSidePadAutoAnnular">
       <property name="text">
        <string>Automatic Annular Ring</string>
       </property>
       <attribute name="buttonGroup">
        <string notr="true">rbtnGroupCmpSidePadShape</string>
       </attribute>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="lblCmpSidePadWarning">
       <property name="minimumSize">
        <size>
         <width>20</width>
         <height>20</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>20</width>
         <height>20</height>
        </size>
       </property>
       <property name="toolTip">
        <string>&lt;b&gt;Use with caution!&lt;/b&gt; If automatic annular rings are enabled for component-side pads, make sure all pads have set the 'component side' property to the correct value. With a wrong configuration, soldering may not be possible due to too small pad area.</string>
       </property>
       <property name="text">
        <string notr="true"/>
       </property>
       <property name="pixmap">
        <pixmap>:/img/status/dialog_warning.png</pixmap>
       </property>
       <property name="scaledContents">
        <bool>true</bool>
       </property>
       <property name="alignment">
        <set>Qt::AlignCenter</set>
       </property>
       <property name="openExternalLinks">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </item>
   <item row="5" column="1" colspan="3">
    <layout class="QHBoxLayout" name="horizontalLayout_3" stretch="0,1">
     <item>
      <widget class="QRadioButton" name="rbtnInnerPadFullShape">
       <property name="text">
        <string>Full Shape</string>
       </property>
       <attribute name="buttonGroup">
        <string notr="true">rbtnGroupInnerPadShape</string>
       </attribute>
      </widget>
     </item>
     <item>
      <widget class="QRadioButton" name="rbtnInnerPadAutoAnnular">
       <property name="text">
        <string>Automatic Annular Ring</string>
       </property>
       <property name="checked">
        <bool>true</bool>
       </property>
       <attribute name="buttonGroup">
        <string notr="true">rbtnGroupInnerPadShape</string>
       </attribute>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>librepcb::editor::UnsignedLengthEdit</class>
   <extends>QWidget</extends>
   <header location="global">librepcb/editor/widgets/unsignedlengthedit.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>librepcb::editor::UnsignedRatioEdit</class>
   <extends>QWidget</extends>
   <header location="global">librepcb/editor/widgets/unsignedratioedit.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>edtStopMaskMaxViaDia</tabstop>
  <tabstop>edtStopMaskClrMin</tabstop>
  <tabstop>edtStopMaskClrRatio</tabstop>
  <tabstop>edtStopMaskClrMax</tabstop>
  <tabstop>edtSolderPasteClrMin</tabstop>
  <tabstop>edtSolderPasteClrRatio</tabstop>
  <tabstop>edtSolderPasteClrMax</tabstop>
  <tabstop>edtPadAnnularRingMin</tabstop>
  <tabstop>edtPadAnnularRingRatio</tabstop>
  <tabstop>edtPadAnnularRingMax</tabstop>
  <tabstop>edtViaAnnularRingMin</tabstop>
  <tabstop>edtViaAnnularRingRatio</tabstop>
  <tabstop>edtViaAnnularRingMax</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>librepcb::editor::BoardDesignRulesDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>257</x>
     <y>386</y>
    </hint>
    <hint type="destinationlabel">
     <x>225</x>
     <y>391</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>librepcb::editor::BoardDesignRulesDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>325</x>
     <y>386</y>
    </hint>
    <hint type="destinationlabel">
     <x>387</x>
     <y>391</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <buttongroups>
  <buttongroup name="rbtnGroupCmpSidePadShape"/>
  <buttongroup name="rbtnGroupInnerPadShape"/>
 </buttongroups>
</ui>
